sound/soc/sof/topology.c

```diff
@@ -1256,15 +1256,35 @@ static int sof_connect_dai_widget(struct snd_soc_component *scomp,
 
 		switch (w->id) {
 		case snd_soc_dapm_dai_out:
-			for_each_rtd_cpu_dai(rtd, i, cpu_dai)
-				cpu_dai->capture_widget = w;
+			for_each_rtd_cpu_dai(rtd, i, cpu_dai) {
```

Comment:
comment on commit message:

typo: wieget

Can you explain "late widget will overwrite the existing BE widget which is not what we want."
I personally have no idea what a late widget is and why the changed loop helps? Did you mean that this function is called multiple times?

Reply:
> @bardliao the change looks good to me I'd modify the commit message to say something like this:
> ASoC: SOF: topology: handle DAI widget connections properly with multiple CPU DAI's
> " Currently, when connecting a DAI widget to the BE CPU DAI, we overwrite the previous connections. This worked because we only ever had 1 CPU DAI for each rtd until now. But with multiple CPU DAI's, a new connection between a BE CPU DAI and the DAI widget should be established without affecting the previous connections. So, modify the loop to set the playback/capture widget for the first BE CPU DAI that does have a connection established previously."

Thanks @ranj063 That's much clear than mine.

Reply:
> comment on commit message:
> 
> typo: wieget
> 
> Can you explain "late widget will overwrite the existing BE widget which is not what we want."
> I personally have no idea what a late widget is and why the changed loop helps? Did you mean that this function is called multiple times?

@plbossart Yes, the function will be called multiple times. For example, assume SDW1-Playback dai link has two cpu dais, and ALH0x102.OUT and ALH0x202.OUT dai widgets are both trying to connect to SDW1-Playback. We will set both SDW1-Playback.cpu_dais[0] and SDW1-Playback.cpu_dais[1]'s playback_widget = ALH0x102.OUT when ALH0x102.OUT is registered. And they will both be overwritten to ALH0x202.OUT when ALH0x202.OUT is registered.